sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -282,6 +289,23 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val DYNAMIC_DATA_PRUNING_ENABLED =
```

Comment:
This looks like the real config in this PR.
`DYNAMIC_DATA_PRUNING_ENABLED` -> `DYNAMIC_COLUMN_PRUNING_ENABLED`?

Reply:
This config is the switch of is the switch of the data column. May be `DYNAMIC_DATA_COLUMN_PRUNING_ENABLED`?
But this is inconsistent with `DYNAMIC_PARTITION_PRUNING_ENABLED`.